File: VB/Q253493/FilterControl/CustomWinFilterTreeNodeModel.vb
Imports DevExpress.Data
Imports DevExpress.XtraEditors.Filtering

Namespace Q253493.FilterControl
    Public Class CustomWinFilterTreeNodeModel
        Inherits WinFilterTreeNodeModel

        Public Sub New(ByVal control As DevExpress.XtraEditors.FilterControl)
            MyBase.New(control)
        End Sub

        Protected Overrides Function CreateDefaultClauseNode([property] As IBoundProperty, currentFilterProperties As IBoundPropertyCollection) As ClauseNode
            Dim result As ClauseNode = MyBase.CreateDefaultClauseNode([property], FilterProperties)
            result.Operation = CType(Control, CustomFilterControl).GetDefaultOperationCore(result.Property, result.Operation)
            Return result
        End Function

        Public Overrides Function CreateClauseNode() As ClauseNode
            Return New CustomClauseNode(Me)
        End Function
    End Class
End Namespace

File: CS/Q253493/FilterControl/CustomWinFilterTreeNodeModel.cs
using DevExpress.Data;
using DevExpress.XtraEditors.Filtering;

namespace Q253493.FilterControl {
    public class CustomWinFilterTreeNodeModel :WinFilterTreeNodeModel {
        public CustomWinFilterTreeNodeModel(DevExpress.XtraEditors.FilterControl control)
            : base(control) { }

        protected override ClauseNode CreateDefaultClauseNode(IBoundProperty property, IBoundPropertyCollection currentFilterProperties)
        {
            ClauseNode result = base.CreateDefaultClauseNode(property, FilterProperties);
            result.Operation = ((CustomFilterControl)Control).GetDefaultOperationCore(result.Property, result.Operation);
            return result;
        }

        public override ClauseNode CreateClauseNode() {
            return new CustomClauseNode(this);
        }
    }
}
